pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -149,13 +162,14 @@ func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s ku
 		return p, &vr, nil
 	case *v1.Pipeline:
 		vr := trustedresources.VerifyResource(ctx, obj, k8s, refSource, verificationPolicies)
-		// Validation of beta fields must happen before the V1 Pipeline is converted into the storage version of the API.
-		// TODO(#6592): Decouple API versioning from feature versioning
-		if err := obj.Spec.ValidateBetaFields(ctx); err != nil {
-			return nil, nil, fmt.Errorf("invalid Pipeline %s: %w", obj.GetName(), err)
+		// Issue a dry-run request to create the remote Pipeline, so that it can undergo validation from validating admission webhooks
+		// without actually creating the Pipeline on the cluster
+		if _, err := tekton.TektonV1().Pipelines(namespace).Create(ctx, obj, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}}); err != nil {
+			if apierrors.IsBadRequest(err) { // Pipeline rejected by validating webhook
```

Comment:
What happens if err is not a BadRequest but a different kind of error (say, InternalError because the Webhook server is down) - it seems like we'd pass through an invalid pipeline in that case?